posthog/hogql_queries/query_runner.py

```diff
@@ -400,9 +400,9 @@ def run(
         fresh_response_dict["timezone"] = self.team.timezone
         fresh_response = CachedResponse(**fresh_response_dict)
 
-        # Dont cache debug queries with errors
+        # Dont cache debug queries with errors and export queries
         has_error: Optional[list] = fresh_response_dict.get("error", None)
-        if has_error is None or len(has_error) == 0:
+        if (has_error is None or len(has_error) == 0) and execution_mode != ExecutionMode.CALCULATION_ALWAYS:
             # TODO: Use JSON serializer in general for redis cache
             fresh_response_serialized = OrjsonJsonSerializer({}).dumps(fresh_response.model_dump())
             cache.set(cache_key, fresh_response_serialized, settings.CACHED_RESULTS_TTL)
```

Comment:
This doesn't seem ideal, because it means we won't cache results even if they're perfectly useful to have in the cache – `ExecutionMode.CALCULATION_ALWAYS` not being limited to exports.

What's the context behind this change BTW? I would think in general exports should still be cached. A shorter TTL maybe, but exporting the same thing a minute apart should not require recalculation.

Reply:
The context is here: https://posthog.slack.com/archives/C02E3BKC78F/p1715693660332099

Reply:
Oh, I see, so Redis mem usage is the concern. I would suggest two changes then:

1. Look for `LimitContext.EXPORT` instead of `ExecutionMode.CALCULATION_ALWAYS`
2. If it's an export, set the cache TTL to something on the order of minutes/hours, rather than the default days
3. 
How does that sound?

Reply:
Fine to adjust this for other cases of the execution mode, but why would we need to cache exports? These are often in the hundreds of MBs and once computed we get the results from s3.

Reply:
Oh right, it can be like a massive list of events. Yeah, given we already effectively cache this in S3, makes sense to bypass the Redis cache for exports